### PR TITLE
feat: add '.decoration-x' utility classes

### PR DESCRIPTION
**Pull request by Techassi**
This PR is intended to be compatible with the new text-decoration syntax of Tailwind v3.

#### Useful resources:

- https://tailwindcss.com/docs/text-decoration
- https://tailwindcss.com/docs/text-decoration-color
- https://tailwindcss.com/docs/text-decoration-style
- https://tailwindcss.com/docs/text-decoration-thickness
- https://tailwindcss.com/docs/text-underline-offset

This PR is a work in progress. I will comment once all new syntax is implemented and ready to be merged.

### CODE REVIEW

**Comment by Techassi** [comment]
### Added Features

#### New default color

The default colors now support `inherit` just like `current (currentColor)` and `transparent`

```css
.text-inherit {
  color: inherit;
}

.bg-inherit {
  background-color: inherit;
}

.decoration-inherit {
  text-decoration-color: inherit;
}
```

#### Text Decoration Type, see [here](https://windicss.org/utilities/typography.html#text-decoration-type)

Now supports another variant 'overline'

```css
.overline {
  -webkit-text-decoration-line: overline;
  text-decoration-line: overline;
}
```

#### Text Decoration Style, see [here](https://windicss.org/utilities/typography.html#text-decoration-style)

Now supports another variant 'wavy'

```css
.decoration-wavy {
  -webkit-text-decoration-style: wavy;
  text-decoration-style: wavy;
}
```

#### Text Decoration Length, see [here](https://windicss.org/utilities/typography.html#text-decoration-length)

See Breaking Changes - Text Decoration Length

### Breaking Changes

#### Text Decoration Color, see [here](https://windicss.org/utilities/typography.html#text-decoration-color)

The current way of setting text decoration colors is:

```css
.underline-green-500 {
  --tw-line-opacity: 1;
  -webkit-text-decoration-color: rgba(16, 185, 129, var(--tw-line-opacity));
  text-decoration-color: rgba(16, 185, 129, var(--tw-line-opacity));
}
```

The new way is:

```css
.decoration-green-500 {
  --tw-text-decoration-opacity: 1;
  -webkit-text-decoration-color: rgba(16, 185, 129, var(--tw-text-decoration-opacity));
  text-decoration-color: rgba(16, 185, 129, var(--tw-text-decoration-opacity));
}
```

#### Text Decoration Opacity, see [here](https://windicss.org/utilities/typography.html#text-decoration-opacity)

This is no longer available/need, as above syntax now support opacity values:

```css
.decoration-green-500\/50 {
  --tw-text-decoration-opacity: 0.5;
  -webkit-text-decoration-color: rgba(16, 185, 129, var(--tw-text-decoration-opacity));
  text-decoration-color: rgba(16, 185, 129, var(--tw-text-decoration-opacity));
}
```

#### Text Decoration Length, see [here](https://windicss.org/utilities/typography.html#text-decoration-length)

The current way of setting the text decoration lenght (thickness) is:

```css
.underline-auto {
  text-decoration-thickness: auto;
}

.underline-8 {
  text-decoration-thickness: 8px;
}
```

The new way is:

```css
.decoration-auto {
  text-decoration-thickness: auto;
}

.decoration-8 {
  text-decoration-thickness: 8px;
}
```

Now supports another variant 'from-font'

```css
.decoration-from-font {
  text-decoration-thickness: from-font;
}
```

### Unchanged

- Text Decoration Offset is unchanged, as Tailwind v3 has the same syntax

File: src/lib/utilities/static.ts [hunk 1]
```diff
@@ -1593,12 +1603,12 @@ export const staticUtilities: StaticUtility = {
...
     'meta': {
       'group': 'textDecoration',
-      'order': 3,
+      'order': 4,
     },
   },
 
   // http://localhost:3001/utilities/typography.html#text-decoration-style
-  'underline-solid': {
+  'decoration-solid': {
```
Comment (antfu) [inline src/lib/utilities/static.ts:1611]: I think we could support both `underline-x` and `decoration-x` at the same time to avoid breaking changes

Comment (Techassi) [reply src/lib/utilities/static.ts:1611]: 2964fff adds `underline-x` fallbacks

Comment (Techassi) [reply src/lib/utilities/static.ts:1611]: b7cea88 adds static `underline-x` fallbacks

**Comment by Techassi** [comment]
I adjusted the test.

One issue I have is that the test are still failing and I'm not quite sure why. Maybe you can point me in the right direction as this is the first time I'm working with tests.

**Comment by antfu** [comment]
You might need to build first and then run the test

**Comment by Techassi** [comment]
Sadly building first does not solve the issue. But I noticed some strange behaviour while looking through the test results. This gets generated by the `Tools / text decoration color & opacity & length & offset / css / 0` processor (`utilities.test.ts` and `utilities.test.ts.yml`):

```css
.decoration-double {
  -webkit-text-decoration-style: double;
  text-decoration-style: double;
.decoration-dashed {
  -webkit-text-decoration-style: dashed;
  text-decoration-style: dashed;
}
.decoration-dotted {
  -webkit-text-decoration-style: dotted;
  text-decoration-style: dotted;
}
.decoration-dashed {
  -webkit-text-decoration-style: dashed;
  text-decoration-style: dashed;
.decoration-double {
  -webkit-text-decoration-style: double;
  text-decoration-style: double;
}
```

Somehow there are missing brackets and the output is doubled.

**Comment by Techassi** [comment]
Today I learned that the order of test results is very important 👀

**Comment by antfu** [comment]
Awesome, thanks a lot!

**Pull request merged**